src/System.Resources.Extensions/src/System/Resources/Extensions/TypeNameComparer.cs

```diff
@@ -0,0 +1,123 @@
+        {
+            if (assemblyQualifiedTypeName1 == null)
+                throw new ArgumentNullException(nameof(assemblyQualifiedTypeName1));
+            if (assemblyQualifiedTypeName2 == null)
+                throw new ArgumentNullException(nameof(assemblyQualifiedTypeName2));
+
+            if (ReferenceEquals(assemblyQualifiedTypeName1, assemblyQualifiedTypeName2))
+                return true;
+
+            ReadOnlySpan<char> typeSpan1 = assemblyQualifiedTypeName1.AsSpan().TrimStart(s_whiteSpaceChars);
+            ReadOnlySpan<char> typeSpan2 = assemblyQualifiedTypeName2.AsSpan().TrimStart(s_whiteSpaceChars);
+
+            // First, compare type names
+            ReadOnlySpan<char> type1 = ReadTypeName(typeSpan1);
+            ReadOnlySpan<char> type2 = ReadTypeName(typeSpan2);
+            if (!type1.Equals(type2, StringComparison.Ordinal))
+                return false;
+
+            // skip separator and whitespace
+            typeSpan1 = typeSpan1.Length > type1.Length ? typeSpan1.Slice(type1.Length + 1).TrimStart(s_whiteSpaceChars) : ReadOnlySpan<char>.Empty;
+            typeSpan2 = typeSpan2.Length > type2.Length ? typeSpan2.Slice(type2.Length + 1).TrimStart(s_whiteSpaceChars) : ReadOnlySpan<char>.Empty;
+
+            // Now, compare assembly simple names ignoring case
+            ReadOnlySpan<char> simpleName1 = ReadAssemblySimpleName(typeSpan1);
+            ReadOnlySpan<char> simpleName2 = ReadAssemblySimpleName(typeSpan2);
+
+            // Don't allow assembly name without simple name portion
+            if (simpleName1.IsEmpty && !typeSpan1.IsEmpty ||
+                simpleName2.IsEmpty && !typeSpan2.IsEmpty)
+                return false;
+
+            // if both are missing assembly name, or either is missing 
+            // assembly name and the other is mscorlib
+            if (simpleName1.IsEmpty)
+                return (simpleName2.IsEmpty || IsMscorlib(simpleName2));
+            if (simpleName2.IsEmpty)
+                return IsMscorlib(simpleName1);
+
+            if (!simpleName1.Equals(simpleName2, StringComparison.OrdinalIgnoreCase))
+                return false;
+
+            // both are mscorlib, ignore culture and key
+            if (IsMscorlib(simpleName1))
+                return true;
+
+            // both have a matching assembly name parse it to get remaining portions
+            // to compare culture and public key token
+            // the following results in allocations.
+            AssemblyName an1 = new AssemblyName(typeSpan1.ToString());
+            AssemblyName an2 = new AssemblyName(typeSpan2.ToString());
+
+            if (an1.CultureInfo?.LCID != an2.CultureInfo?.LCID)
+                return false;
+
+            byte[] pkt1 = an1.GetPublicKeyToken();
+            byte[] pkt2 = an2.GetPublicKeyToken();
+            return pkt1.AsSpan().SequenceEqual(pkt2);
+        }
+
+        public int GetHashCode(string assemblyQualifiedTypeName)
```

Comment:
Nit: static?

Reply:
Oh, nevermind, it's part of the interface.

Reply:
Cannot be static since it is implementing IEqualityComparer<string>